### ISSUE TITLE
Rollout `return_async_id` to more functions

### ISSUE THREAD

**Issue by MariusWirtz** [cubewise-code/tm1py#1132]
I propose we roll out the `return_async_id` argument to more functions. 

It would be helpful to have it on tasks that are "single request based" but yet can long to complete, such as:

- `add_elements`
- `add_edges`

**Comment by rclapp** [comment]
Can we add it as a decorator? Also what about queries?

On Jul 3, 2024 3:52 PM, Marius Wirtz ***@***.***> wrote:

I propose we roll out the return_async_id argument to more functions.

It would be helpful to have it on tasks that are "single request based" but yet can long to complete, such as:

  *   add_elements
  *   add_edges

**Comment by MariusWirtz** [comment]
Yes. The preferred solution would be to solve it with a generic decorator that we could use on all functions.

To limit the complexity of that decorator, we could, in a first version, limit it to functions that are single request-based
